### ISSUE TITLE
Militia use sector equipment is broken, militia will spawn with lose all it's previously equipped guns after a day ingame

### ISSUE THREAD

**Issue by MeshVoid** [1dot13/source#99]
Version: latest build 57fcdaf.

It seems that feature to enable militia use sector equipment is broken and not working as it is supposed to. 

I am referring to this particular option in Ja2_Options.ini:
> 
> if set to true, militia will take parts of its equipment from a sector's inventory.
> MILITIA_USE_SECTOR_EQUIPMENT                        = TRUE

The way it's supposed to work, according to feature description, is that militia will pick up guns and armor from sector inventory unless you Tab + Lmb click on a particular item to be avoided by militia. This a very interesting feature and makes arming militia more fun, but sadly it doesn't work as it is supposed to and costed me several failed runs, until I have identified that my militia was just ditching items and getting naked after a while resulting in lost combat engagements.

According to description on forums and in .ini file this feature makes militia pick items from sector inventory up after they have spawned when you've finished training them. They actually do pickup items successfully, but after a while in game, they get rid of all items and engage in fighting completely naked and essentially worthless, making militia completely useless for guard duty.

Here are some screenshots:
![image](https://user-images.githubusercontent.com/35863328/213646673-ef79aa01-44f8-4aff-b671-1802b7e6d62e.png)
Right after spawning Militia is armed and they've picked up most of the guns that were lying around in sector inventory.

Day has passed in game, the same militia operatives (no new militia were trained or spawned) are now naked:
![image](https://user-images.githubusercontent.com/35863328/213647488-b57dcd17-cb26-46a0-aaee-c29d0aa3a7a2.png)

Naturally, it gets quite ugly during the fights and you expect your militia armed with weapons you left them:
![image](https://user-images.githubusercontent.com/35863328/213647689-af04af3e-a704-426f-8c2f-82c0a53d26a1.png)

This feature supposed to make equipping militia harder but potentially making them deadlier if equipped with superior gear, but because they ditch items, it's broken and militia is useless when you play with this option.

I have also attached archive with 2 savegames to reproduce the error. 

Savegame04.sav > Militia successfully equipped weapons after training.

Savegame05.save > A day in-game has passed, militia has no items equipped and is running around naked. 

[SavedGames.zip](https://github.com/1dot13/source/files/10464416/SavedGames.zip)

**Comment by Asdow** [comment]
The militia should be picking up weapons from sector inventory just before a battle and then dropping them back to sector afterwards. Not having weapons when there is not a battle going on should not be out of normal but they definitely should be equipped during combat.

Do you have the setting enabled that they also need ammunition from sector inventory, or that they spawn their own ammo?

**Comment by MeshVoid** [comment]
> The militia should be picking up weapons from sector inventory just before a battle and then dropping them back to sector afterwards. Not having weapons when there is not a battle going on should not be out of ordinary but they definitely should be equipped during combat.

Yup, I just realized that it was written in the description in .ini file:

> ; Once a sector gets unloaded, the militia will drop the equipment taken again into the sector.

What I noticed is that militia members take guns and ammo, and equip them once, but after the sector is unloaded those guns never return back to sector inventory, they're gone forever and militia is naked now too.

Yup, I had this option turned on:
> MILITIA_USE_SECTOR_EQUIPMENT_AMMO                   = TRUE
And accordingly left ammo for every gun in sector inventory for militia to pick up.

**Comment by Asdow** [comment]
I still haven't been able to reproduce this. Does the militia drop any equipment if you do CTRl + . and press "militia inspection" button? They should drop all their current equipment to the ground
